api/streamableResponseWriter.go

```diff
@@ -20,3 +23,12 @@ func (wf *streamableResponseWriter) Write(b []byte) (n int, err error) {
 	wf.Flush()
 	return n, err
 }
+
+type closeFunc = func() error
+
+func streamCleanup(ctx context.Context, desc string, closer closeFunc) {
```

Comment:
We should also wrap streamableResponseWriter's  `Write` method to ensure that nothing's written on a cancelled context. This would eliminate https://github.com/puppetlabs/wash/blob/master/api/exec.go#L18-L28. Then, all of this context-cancellation stuff can be baked into `newStreamableResponseWriter(ctx, w, endpoint, cleanup)` (I like cleanup better b/c it's more generic). The goroutine can go inside `new`, like in `NewExecCommand`.

Reply:
I think combining them makes it more awkward. And guarding Write isn't really needed because we'll get back an error when we try to write anyway (the write will have already been closed).

Reply:
I added the `sendPacket` helper in the Exec endpoint because I noticed the server logs were cluttered with context-cancellation related errors. Having the guarded write is nice b/c it guarantees that we avoid redundant error logging for all of our endpoints. I also think it is reasonable for a streamableResponseWriter's Write method to noop on a cancelled context.

I can understand not combining the cleanup func though. It is weird to think of a streamableResponseWriter as handling non-response related cleanup.  Would still be nice if `closer` => `cleanup` to reflect `streamCleanup`'s genericness.